### ISSUE TITLE
Connecting front and back end logic for search bar

### ISSUE THREAD

**Issue by hibaabdullahhamad** [CMU-17313Q/nodebb-f24-syntax_squad#25]
This issue will connect the front and back end logic for the search bar feature, it will ensure that once a user searches a keyword in the search bar, the relevant function (created in issue #20) is called. In this issue, we will work on displaying the resulting messages of the searched keyword. A separate issue has been created for front-end rendering, issue #70, as this issue is too big.

**Estimated effort:**
The task will require about 1-2 days to complete. Bugs and other issues might take another day to fix. Around 2 days in total.

**Dependencies among other tasks:**
This task depends on issues #19 and #20.

**Initial assignments for team members:**
Initially assigned to Hiba Hamad, @hibaabdullahhamad, officially assigned to Najoud Al-Talib, @njouud 

**Milestones to indicate which sprint the issue should be completed in:**
Sprint 1

**Justification for not meeting milestone:**
Finding files took longer than expected for the back end and front end implementations, as well as independently testing front end. Worked on the wrong files for the back end at first, delaying the completion further. 

**Acceptance criteria:**
This feature will be tested using the keyword "spanish" in the search bar. If the search bar works, NodeBB will return 5 out of the 15 added discussions that have the word "spanish". This will be tested with other key terms as well as other accounts to ensure consistency. **Sprint 2 development: testing only in the console of developer tools, front-end rendering is for issue #70.**
